src/Servers/Kestrel/Core/src/Internal/Http/Http1Connection.cs

```diff
@@ -628,6 +627,7 @@ protected override void OnReset()
             // Reset Http1 Features
             _currentIHttpMinRequestBodyDataRateFeature = this;
             _currentIHttpMinResponseDataRateFeature = this;
+            _currentIPersistentStateFeature = ConnectionFeatures.Get<IPersistentStateFeature>();
```

Comment:
I added `IPersistentStateFeature` as a known feature on `HttpProtocol`. `Http2Stream` needs to customize how it is used so we persist per-stream, not per-connection for HTTP/2. Meanwhile, HTTP/1.1 and HTTP/3 can just reference the state directly from the transport.

Advice here?

Also, I need to unit test that state is persisted for HTTP/1.1. Where are unit tests that test that HttpContext is reused between HTTP/1.1 requests?

Reply:
`_currentIPersistentStateFeature = ConnectionFeatures.Get<IPersistentStateFeature>();`

Actually, is this necessary? I think the feature lookup will fallback to getting the feature from the connection anyway.